File: frontend/src/models/product.ts
export type Location = 'JO' | 'SA'

export interface Product {
  id: number
  title: string
  description: string | null
  price: string
  location: Location
  created_at: string
  updated_at: string
}

export interface Pagination {
  page: number
  per_page: number
  total: number
  pages: number
}

export interface ProductListResponse {
  items: Product[]
  pagination: Pagination
}

File: backend/app/schemas/product.py
from marshmallow import Schema, fields


class ProductSchema(Schema):
    id = fields.Int(dump_only=True)
    title = fields.Str(dump_only=True)
    description = fields.Str(dump_only=True, allow_none=True)
    price = fields.Str(dump_only=True)
    location = fields.Str(dump_only=True)
    created_at = fields.Str(dump_only=True)
    updated_at = fields.Str(dump_only=True)


class PaginationSchema(Schema):
    page = fields.Int(required=True)
    per_page = fields.Int(required=True)
    total = fields.Int(required=True)
    pages = fields.Int(required=True)


class ProductListResponseSchema(Schema):
    items = fields.List(fields.Nested(ProductSchema), required=True)
    pagination = fields.Nested(PaginationSchema, required=True)
